### PR TITLE
Adding Screen to Search/Filter Metadata

### PR DESCRIPTION
**Pull request by jchavez137**
- [X] Created New entry point for Search/Filter Metadata
- [X] Reorganized entry points for within a Menu
- [X] Created table components to display results for Metadata search/Filter
- [X] Added sample result display for static search

image
image

### CODE REVIEW

File: ui/src/app/metadata/detail/metadata-list/metadata-list.component.ts [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+    new ApiNamedReference({id: "id2", framework: "framework2", name: "name2", type: MetadataType.Category, value: "value2"}),
+    new ApiNamedReference({id: "id3", framework: "framework3", name: "name3", type: MetadataType.Category, value: "value3"}),
+    new ApiNamedReference({id: "id4", framework: "framework4", name: "name4", type: MetadataType.Category, value: "value4"}),
+    new ApiNamedReference({id: "id5", framework: "framework5", name: "name5", type: MetadataType.Category, value: "value5"}),
+    new ApiNamedReference({id: "id6", framework: "framework6", name: "name6", type: MetadataType.Category, value: "value6"}),
+    new ApiNamedReference({id: "id7", framework: "framework7", name: "name7", type: MetadataType.Category, value: "value7"}),
+    new ApiNamedReference({id: "id8", framework: "framework8", name: "name8", type: MetadataType.Category, value: "value8"}),
+  ], 8)
+
+  results: PaginatedMetadata = this.sampleJobCodeResult
```
Comment (manuel-delvillar) [inline ui/src/app/metadata/detail/metadata-list/metadata-list.component.ts:62]: The default value in select is a category (named reference), so this line should be results: PaginatedMetadata = this.sampleNamedReferenceResult

File: ui/src/app/metadata/detail/metadata-list/metadata-list.component.ts [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+
+  clearSelectedItemsFromTable = new Subject<void>()
+  constructor(protected authService: AuthService) {
+    super()
+    this.typeControl.valueChanges.subscribe(
```
Comment (manuel-delvillar) [inline ui/src/app/metadata/detail/metadata-list/metadata-list.component.ts:67]: Probably move `this.typeControl.valueChanges.subscribe` to a method and call the method on the constructor.

File: ui/src/app/metadata/IMetadata.ts [hunk 3]
```diff
@@ -0,0 +1,25 @@
+import {ApiJobCode} from "./job-codes/Jobcode"
+import {ApiNamedReference} from "./named-references/NamedReference"
+
+export interface IMetadata {
+  apiJobCodes?: ApiJobCode[],
+  apiNamedReferences?: ApiNamedReference[]
+}
+
+export class ApiMetadata implements IMetadata {
```
Comment (manuel-delvillar) [inline ui/src/app/metadata/IMetadata.ts:9]: Seems `IMetadata` and `ApiMetadata` are unused, but if there is plan to use them in the implementation ignore this comment.

File: ui/src/app/metadata/IMetadata.ts [hunk 4]
```diff
@@ -0,0 +1,25 @@
+import {ApiJobCode} from "./job-codes/Jobcode"
+import {ApiNamedReference} from "./named-references/NamedReference"
```
Comment (manuel-delvillar) [inline ui/src/app/metadata/IMetadata.ts:2]: According to code style in other files, seems we need white space in imports.

**Review by manuel-delvillar — COMMENTED** [review]
Maybe we need to add "RSD Metadata" to the mobile view.

File: ui/src/app/metadata/detail/metadata-list/metadata-list.component.ts (hunk 1 above)
Comment (jchavez137) [reply ui/src/app/metadata/detail/metadata-list/metadata-list.component.ts:62]: nice catch, yeah I forgot to check on the default selection.

File: ui/src/app/metadata/detail/metadata-list/metadata-list.component.ts (hunk 2 above)
Comment (jchavez137) [reply ui/src/app/metadata/detail/metadata-list/metadata-list.component.ts:67]: Right, I will put it into a OnInit implementation.

File: ui/src/app/metadata/IMetadata.ts (hunk 3 above)
Comment (jchavez137) [reply ui/src/app/metadata/IMetadata.ts:9]: Yeah, I will rename the file to match with the only class used here, and let that code in this very file for future code inclusion.

File: ui/src/app/metadata/IMetadata.ts (hunk 4 above)
Comment (jchavez137) [reply ui/src/app/metadata/IMetadata.ts:2]: Correcting..

File: ui/package.json [hunk 5]
```diff
@@ -17,10 +17,12 @@
   "private": true,
   "dependencies": {
     "@angular/animations": "~12.2.17",
+    "@angular/cdk": "~12.2.13",
```
[the remark this answers is not in the document]
Comment (wgu-edwin) [reply ui/package.json:20]: i agree, i think it should be

[the remark this answers is not in the document]
Comment (jchavez137) [reply ui/package.json]: Moved to dev dependencies

**Review by manuel-delvillar — APPROVED** [review]
Looks ready to merge

**Pull request merged**